packages/shared/profiles/ProfileAsPromise.ts

```diff
@@ -11,53 +11,75 @@ const PROFILE_SOFT_TIMEOUT_MS = 5000
 // We reject the profile promise if more time than this has passed
 const PROFILE_HARD_TIMEOUT_MS = COMMS_PROFILE_TIMEOUT + 20000
 
-export function ProfileAsPromise(userId: string, version?: number, profileType?: ProfileType): Promise<Avatar> {
+export function ProfilesAsPromise(userIds: string[], version?: number, profileType?: ProfileType): Promise<Avatar[]> {
```

Comment:
For clarity purposes and to prevent accidents with accesors, I'd recommend to have two versions of this function. The new `ProfilesAsPromise(): Array<Profile>` and the original `ProfileAsPromise(): Profile`. That will simplify a lot the changes required to test this PR